app/models/site_channel_details.rb

```diff
@@ -8,6 +8,8 @@ class SiteChannelDetails < ApplicationRecord
   # formats to support more publishers.
   validates :brave_publisher_id, uniqueness: { if: -> { !errors.include?(:brave_publisher_id_unnormalized) && brave_publisher_id.present? && brave_publisher_id_changed? && verified_publisher_id_exists? } }
 
+  validate :same_publisher_brave_publisher_id_unique, if: -> { !errors.include?(:brave_publisher_id_unnormalized) && brave_publisher_id.present? && brave_publisher_id_changed? }
```

Comment:
Uniqueness constraint should be done in database, not in Rails validation, as a multi-process system cannot guarantee uniqueness due to non-shared memory.